php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1402 @@
+        .is(f.emptyStatement(b.token(PHPPunctuator.SEMICOLON)));
+  }
+
+  public ReturnStatementTree RETURN_STATEMENT() {
+    return b.<ReturnStatementTree>nonterminal(PHPLexicalGrammar.RETURN_STATEMENT)
+        .is(f.returnStatement(b.token(PHPKeyword.RETURN), b.optional(EXPRESSION()), EOS()));
+  }
+
+  public ContinueStatementTree CONTINUE_STATEMENT() {
+    return b.<ContinueStatementTree>nonterminal(PHPLexicalGrammar.CONTINUE_STATEMENT)
+        .is(f.continueStatement(b.token(PHPKeyword.CONTINUE), b.optional(EXPRESSION()), EOS()));
+  }
+
+  public BreakStatementTree BREAK_STATEMENT() {
+    return b.<BreakStatementTree>nonterminal(PHPLexicalGrammar.BREAK_STATEMENT)
+        .is(f.breakStatement(b.token(PHPKeyword.BREAK), b.optional(EXPRESSION()), EOS()));
+  }
+
+  public TryStatementTree TRY_STATEMENT() {
+    return b.<TryStatementTree>nonterminal(PHPLexicalGrammar.TRY_STATEMENT)
+        .is(f.tryStatement(
+            b.token(PHPKeyword.TRY),
+            BLOCK(),
+            b.zeroOrMore(CATCH_BLOCK()),
+            b.optional(f.newTuple2(b.token(PHPKeyword.FINALLY), BLOCK()))));
+  }
+
+  public CatchBlockTree CATCH_BLOCK() {
+    return b.<CatchBlockTree>nonterminal(PHPLexicalGrammar.CATCH_BLOCK)
+        .is(f.catchBlock(
+            b.token(PHPKeyword.CATCH),
+            b.token(PHPPunctuator.LPARENTHESIS),
+            NAMESPACE_NAME(),
+            b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER),
+            b.token(PHPPunctuator.RPARENTHESIS),
+            BLOCK()
+        ));
+
+  }
+
+  public BlockTree BLOCK() {
+    return b.<BlockTree>nonterminal(PHPLexicalGrammar.BLOCK)
+        .is(f.block(
+            b.token(PHPPunctuator.LCURLYBRACE),
+            b.zeroOrMore(INNER_STATEMENT()),
+            b.token(PHPPunctuator.RCURLYBRACE)
+        ));
+  }
+
+  public GotoStatementTree GOTO_STATEMENT() {
+    return b.<GotoStatementTree>nonterminal(PHPLexicalGrammar.GOTO_STATEMENT)
+        .is(f.gotoStatement(b.token(PHPKeyword.GOTO), b.token(PHPLexicalGrammar.IDENTIFIER), EOS()));
+  }
+
+  public ExpressionStatementTree EXPRESSION_STATEMENT() {
+    return b.<ExpressionStatementTree>nonterminal(PHPLexicalGrammar.EXPRESSION_STATEMENT)
+        .is(f.expressionStatement(EXPRESSION(), EOS()));
+  }
+
+  public LabelTree LABEL() {
+    return b.<LabelTree>nonterminal(PHPLexicalGrammar.LABEL)
+        .is(f.label(b.token(PHPLexicalGrammar.IDENTIFIER), b.token(PHPPunctuator.COLON)));
+  }
+
+  public InternalSyntaxToken EOS() {
+    return b.<InternalSyntaxToken>nonterminal(PHPLexicalGrammar.EOS)
+        .is(b.firstOf(b.token(PHPPunctuator.SEMICOLON), b.token(PHPLexicalGrammar.INLINE_HTML)));
+  }
+
+  /**
+   * [ END ] Statement
+   */
+
+  public ExpressionTree EXPRESSION() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.EXPRESSION)
+      .is(f.expression(b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER)));
+  }
+
+  /**
+   * [ START ] Expression
+   */
+
+  // FIXME Fake implementation
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)